### PR TITLE
add validator rewards pools

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 infinite pools for rewards don't exist for validator regime

 #### Summary of Changes
 * define pools' location
 * add pools to genesis

Fixes #

### CODE REVIEW

File: programs/stake_api/src/rewards_pools.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        builder = builder.rewards_pool(
+            pubkey,
+            Account::new_data(std::u64::MAX, &StakeState::RewardsPool, &crate::id()).unwrap(),
+        );
+        pubkey = Pubkey::new(hash(pubkey.as_ref()).as_ref());
+    }
+    builder
+}
+
+pub fn random_id() -> Pubkey {
```
Comment (aeyakovenko) [inline programs/stake_api/src/rewards_pools.rs]: is this just used in tests?

Comment (rob-solana) [reply programs/stake_api/src/rewards_pools.rs]: no, will be picked up by stake_instruction::redeem_vote_credits()

Comment (aeyakovenko) [reply programs/stake_api/src/rewards_pools.rs]: I thought we had a plan to avoid random pools?

Comment (rob-solana) [reply programs/stake_api/src/rewards_pools.rs]: this is the way.  the pools are pre-determined and all infinite, and there are 256 of 'em.  you redeem from one at random.

Comment (aeyakovenko) [reply programs/stake_api/src/rewards_pools.rs]: @rob-solana so this is the client?  or inside ether bank?

Comment (rob-solana) [reply programs/stake_api/src/rewards_pools.rs]: the client runs this routine

**Pull request merged**